src/UI/processmonitor.cpp

```diff
@@ -1,89 +1,61 @@
 #include "pch.h"
 #include "processmonitor.h"
 #include "processesseeker.h"
+#include "processesstorage.h"
+
+ProcessMonitor::ProcessMonitor(IProcessesSeeker* processesSeeker)
+               :processesStorage{std::make_unique<ProcessesStorage>()},
+               processesSeeker{processesSeeker}{};
 
 void ProcessMonitor::updateProcessesTable() {
-    std::vector<ProcessInfo> updatedProcesses = processesSeeker->getSystemProcesses();
-    std::sort(updatedProcesses.begin(),updatedProcesses.end());
-    processesInfo = mergeProcessesSortedLists(processesInfo, updatedProcesses);
-    qDebug() << "Processes size "<< QVariant(processesInfo.size()) << "updatedProcesses size" << QVariant(updatedProcesses.size());
+    auto update = processesSeeker->getSystemProcesses();
+    processesStorage->update(update);
 }
 
 ProcessInfo ProcessMonitor::getCopyOfProcessInfoByIndex(const size_t Index) const{
-    return processesInfo[Index];
+    return processesStorage->getProcessByIndex(Index);
 }
 
 ProcessInfo ProcessMonitor::getCopyOfProcessInfoByPid(const DWORD Pid) const{
-    return getCopyOfProcessInfoByIndex(getIndexOfProcessBySpecyficPid(Pid));
+    return processesStorage->getProcessByPid(Pid);
 }
 
-void ProcessMonitor::setProcessEditableFieldByIndex(const size_t Index, const ProcessEditableFields field, const bool value){
+void ProcessMonitor::setProcessEditableField(ProcessInfo& process, const ProcessEditableFields field, const bool value){
     switch(field){
     case ProcessEditableFields::readPerm:
-        processesInfo[Index].readPermission = value;
+        process.readPermission = value;
     break;
     case ProcessEditableFields::writePerm:
-        processesInfo[Index].writePermission = value;
+        process.writePermission = value;
     break;
     case ProcessEditableFields::openPerm:
-        processesInfo[Index].openPermission = value;
+        process.openPermission = value;
     break;
     case ProcessEditableFields::deletePerm:
-        processesInfo[Index].deletePermission = value;
+        process.deletePermission = value;
     break;
     case ProcessEditableFields::isMonitored:
-        processesInfo[Index].isMonitored = value;
+        process.isMonitored = value;
     break;
     case ProcessEditableFields::isDllInjected:
-        processesInfo[Index].isDllInjected = value;
+        process.isDllInjected = value;
     break;
     }
```

Comment:
Then you should send this updated data to the hook. Please add a corresponding //TODO comment here or a call of the stub function with //TODO comment.

Reply:
I added the TODO comment.